core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -199,7 +201,7 @@ public void set(PropertyKey key, Object value) {
    * @param value the value for the key
    * @param source the source of the the properties (e.g., system property, default and etc)
    */
-  public void set(PropertyKey key, Object value, Source source) {
+  public void set(@Nonnull PropertyKey key, @Nonnull Object value, @Nonnull Source source) {
```

Comment:
this annotation is required?

Reply:
it's required or else spotbugs complains about possible null derefence.

AFAIK it can't determine that `Preconditions` does the null check